src/main.ts

```diff
@@ -1382,3 +1382,25 @@ export class SqlOpsDataClient extends LanguageClient {
 		});
 	}
 }
+
+/**
+ * Base class containing shared code to reduce boilerplate for services
+ */
+export abstract class BaseService {
+	constructor(protected readonly client: SqlOpsDataClient) { }
+
+	/**
+	 * Sends the specified request and logs the failure through the client if one occurs before rethrowing.
+	 * @param type RequestType, typically in the format 'contracts.DoThingRequest.type'
+	 * @param params parameters to be passed to the request
+	 * @returns result from the request
+	 */
+	protected runWithErrorHandling<P, R, E, RO>(type: RequestType<P, R, E, RO>, params: P): Thenable<R> {
+		try {
+			return this.client.sendRequest(type, params);
```

Comment:
if you are not awaiting, then the try/catch block won't work. should use .then(result=>{}, error=>{})